tests/test_sorted_results.py

```diff
@@ -162,15 +173,20 @@ def test_add_results_failing_to_passing(self):
         and then a measurement makes one of the results passing
         """
 
+        # Create constraint_manager for all model names
+        constraints = {
+            "profile_models": {
+                str(i): {"constraints": {'perf_throughput': {'min': 1000}}}
+                for i in sample(range(10), 10)
+            }
+        }
+        constraint_manager = construct_constraint_manager(constraints)
+
         # Create 10 failing results
         avg_gpu_metrics = {0: {'gpu_used_memory': 6000, 'gpu_utilization': 60}}
         for i in sample(range(10), 10):
             model_name = str(i)
             model_config_names = [f"{model_name}_config_0"]
-            constraints = {
-                model_name: {'perf_throughput': {'min': 1000}}
-            }
-            constraint_manager = construct_constraint_manager(constraints=constraints)
```

Comment:
This is the type code I was referring to about only creating one constraint_manager for the test, instead of creating a new one over and over. It looks like this has been fixed.

Reply:
Yes, it's fixed now 👍